contracts/amm/AmmOpenSwapService.sol

```diff
@@ -0,0 +1,819 @@
+        require(usdtPool.ammStorage != address(0), string.concat(IporErrors.WRONG_ADDRESS, " USDT pool ammStorage"));
+        require(usdtPool.ammTreasury != address(0), string.concat(IporErrors.WRONG_ADDRESS, " USDT pool ammTreasury"));
+
+        require(usdcPool.asset != address(0), string.concat(IporErrors.WRONG_ADDRESS, " USDC pool asset"));
+        require(usdcPool.ammStorage != address(0), string.concat(IporErrors.WRONG_ADDRESS, " USDC pool ammStorage"));
+        require(usdcPool.ammTreasury != address(0), string.concat(IporErrors.WRONG_ADDRESS, " USDC pool ammTreasury"));
+
+        require(daiPool.asset != address(0), string.concat(IporErrors.WRONG_ADDRESS, " DAI pool asset"));
+        require(daiPool.ammStorage != address(0), string.concat(IporErrors.WRONG_ADDRESS, " DAI pool ammStorage"));
+        require(daiPool.ammTreasury != address(0), string.concat(IporErrors.WRONG_ADDRESS, " DAI pool ammTreasury"));
+
+        require(iporOracle != address(0), string.concat(IporErrors.WRONG_ADDRESS, " iporOracle"));
+        require(
+            iporRiskManagementOracle != address(0),
+            string.concat(IporErrors.WRONG_ADDRESS, " iporRiskManagementOracle")
+        );
+        require(spreadRouter != address(0), string.concat(IporErrors.WRONG_ADDRESS, " spreadRouter"));
+
+        _iporPublicationFee = configuration.iporPublicationFee;
+        _maxSwapCollateralAmount = configuration.maxSwapCollateralAmount;
+
+        _liquidationDepositAmount = configuration.liquidationDepositAmount;
+        _liquidationDepositAmountWad = configuration.liquidationDepositAmount * Constants.D18;
+
+        _minLeverage = configuration.minLeverage;
+        _maxLeverage = configuration.maxLeverage;
+
+        _openingFeeRate = configuration.openingFeeRate;
+        _openingFeeTreasuryPortionRate = configuration.openingFeeTreasuryPortionRate;
+
+        _usdt = usdtPool.asset;
+        _usdtAmmStorage = usdtPool.ammStorage;
+        _usdtAmmTreasury = usdtPool.ammTreasury;
+
+        _usdc = usdcPool.asset;
+        _usdcAmmStorage = usdcPool.ammStorage;
+        _usdcAmmTreasury = usdcPool.ammTreasury;
+
+        _dai = daiPool.asset;
+        _daiAmmStorage = daiPool.ammStorage;
+        _daiAmmTreasury = daiPool.ammTreasury;
+
+        _iporOracle = iporOracle;
+        _iporRiskManagementOracle = iporRiskManagementOracle;
+        _spreadRouter = spreadRouter;
+    }
+
+    function openSwapPayFixed28daysUsdt(
+        address onBehalfOf,
+        uint256 totalAmount,
+        uint256 acceptableFixedInterestRate,
+        uint256 leverage
+    ) external override returns (uint256) {
+        Context memory context = Context({
+            onBehalfOf: onBehalfOf,
+            asset: _usdt,
+            ammStorage: _usdtAmmStorage,
+            ammTreasury: _usdtAmmTreasury,
+            maturity: 28,
+            spreadMethodSig: "calculatePayFixed28Days((address,uint256,uint256,uint256,int256,uint256,uint256,uint256,uint256,uint256))"
```

Comment:
will it be replaced by signatures from interface later?

Reply:
Yes, when interface came with Spread implementation